portability-core/src/main/java/org/dataportabilityproject/serviceProviders/microsoft/MicrosoftServiceProvider.java

```diff
@@ -73,24 +74,30 @@
     }
 
     @Override
-    public OfflineAuthDataGenerator getOfflineAuthDataGenerator(PortableDataType dataType) {
+    public OfflineAuthDataGenerator getOfflineAuthDataGenerator(PortableDataType dataType, ServiceMode serviceMode) {
         switch (dataType) {
             case CALENDAR:
+                // IMPORT and EXPORT are supported for CALENDAR
                 return microsoftAuth;
             case MAIL:
+                // Only EXPORT is supported for MAIL
```

Comment:
do we purposely not implement import or just not yet? can you document / add a TODO to implement?

Reply:
I think its purposefully. I'll refactored the way we figure out if a service is supported or and added a TODO for import
PTAL.